### PR TITLE
[SHELL32] Implement 'Open file location' of shortcut files

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
JIRA issue: [CORE-12770](https://jira.reactos.org/browse/CORE-12770)
"Open file location" is a feature to open the location of the target of a shortcut file.

open-loc

### CODE REVIEW

File: dll/win32/shell32/CShellLink.cpp [hunk 1]
```diff
@@ -2578,39 +2578,64 @@ HRESULT STDMETHODCALLTYPE CShellLink::Initialize(PCIDLIST_ABSOLUTE pidlFolder, I
...
-HRESULT STDMETHODCALLTYPE CShellLink::InvokeCommand(LPCMINVOKECOMMANDINFO lpici)
+HRESULT CShellLink::DoOpenFileLocation()
 {
-    LPWSTR args = NULL;
-    LPWSTR path = NULL;
+    WCHAR szParams[MAX_PATH + 64];
+    StringCbPrintfW(szParams, sizeof(szParams), L"/select,%s", m_sPath);
+
+    INT_PTR ret;
+    ret = reinterpret_cast<INT_PTR>(ShellExecuteW(NULL, NULL, L"explorer.exe", szParams,
```
Comment (learn-more) [inline dll/win32/shell32/CShellLink.cpp]: Ideally we should probably use `SHOpenFolderAndSelectItems` here, but that is not 100% implemented in ros yet...

Comment (katahiromz) [reply dll/win32/shell32/CShellLink.cpp]: Can `explorer.exe` select multiple items?

Comment (learn-more) [reply dll/win32/shell32/CShellLink.cpp]: using `SHOpenFolderAndSelectItems` it should be able to.

Comment (Mouvedia) [reply dll/win32/shell32/CShellLink.cpp]: check CORE-9367

**Comment by ale5000-git** [comment]
@katahiromz
The code seems to hardcode explorer.exe but on Windows there is a registry settings to change the shell if I'm not wrong.
Isn't it possible to support it?

**Comment by katahiromz** [comment]
Key name: HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon
Value name: Shell

**Comment by ale5000-git** [comment]
Thanks, but actually I was meaning supporting it here: https://github.com/reactos/reactos/pull/2028/files#diff-c8c0c634fe8ba43f11647ffe95ec7036R2626

**Comment by katahiromz** [comment]
https://jira.reactos.org/browse/CORE-16500

**Pull request merged**